data-serving/data-service/src/controllers/case.ts

```diff
@@ -181,15 +181,34 @@ const findCaseIdsWithCaseReferenceData = async (
                 'caseReference.sourceEntryId': c.caseReference.sourceEntryId,
             };
         });
-    const result =
-        providedCaseReferenceData.length > 0
-            ? await Case.find()
-                  .select('_id')
-                  .or(providedCaseReferenceData)
-                  .lean()
-                  .exec()
-            : [];
-    return result.map((res) => String(res['_id']));
+
+    return providedCaseReferenceData.length > 0
+        ? Case.find()
+              .or(providedCaseReferenceData)
+              .select({ _id: 1, caseReference: 1, revisionMetadata: 1 })
```

Comment:
Did the `lean()` not make a difference, here? I admit I didn't conclusively check that it altered query behavior if a `select` clause was present. I expect it did, and that with `lean()`, this won't return a `CaseDocument`.

Reply:
I've run it a few times and it doesn't seem to have helped (if anything, it made it a little worse -- though I am not doing this very scientifically!)

Since it doesn't seem to help much, I think I'd prefer to keep it in for the type safety. WDYT?

Reply:
SG. FMI, does it return a `CaseDocument` with only the `select` fields populated?

Reply:
yep exactly!